rules/Arguments/Rector/MethodCall/SwapMethodCallArgumentsRector.php

```diff
@@ -0,0 +1,138 @@
+{
+    /**
+     * @var string
+     */
+    private const JUST_SWAPPED = 'just_swapped';
+
+    /**
+     * @var SwapMethodCallArguments[]
+     */
+    private array $methodArgumentSwaps = [];
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Reorder arguments in method calls', [
+            new ConfiguredCodeSample(<<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function run(Caller $caller)
+    {
+        return $caller->call('one', 'two', 'three');
+    }
+}
+CODE_SAMPLE
+                , <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function run(Caller $caller)
+    {
+        return $caller->call('three', 'two', 'one');
+    }
+}
+CODE_SAMPLE
+                , [new SwapMethodCallArguments('Caller', 'call', [2, 1, 0])]),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param MethodCall|StaticCall $node
+     */
+    public function refactor(Node $node): MethodCall|StaticCall|null
+    {
+        $isJustSwapped = (bool) $node->getAttribute(self::JUST_SWAPPED, false);
+        if ($isJustSwapped) {
+            return null;
+        }
+
+        foreach ($this->methodArgumentSwaps as $methodArgumentSwap) {
+            if (! $this->isName($node->name, $methodArgumentSwap->getMethod())) {
+                continue;
+            }
+
+            if (! $this->isObjectTypeMatch($node, $methodArgumentSwap->getObjectType())) {
+                continue;
+            }
+
+            $newArguments = $this->resolveNewArguments($methodArgumentSwap, $node);
+            if ($newArguments === []) {
+                return null;
+            }
+
+            foreach ($newArguments as $newPosition => $argument) {
+                $node->args[$newPosition] = $argument;
+            }
+
+            $node->setAttribute(self::JUST_SWAPPED, true);
+
+            return $node;
+        }
+        return null;
+    }
+
+    public function configure(array $configuration): void
+    {
+        Assert::allIsAOf($configuration, SwapMethodCallArguments::class);
+        $this->methodArgumentSwaps = $configuration;
+    }
+
+    private function isObjectTypeMatch(MethodCall|StaticCall $node, ObjectType $objectType): bool
+    {
+        if ($node instanceof MethodCall) {
+            return $this->isObjectType($node->var, $objectType);
+        }
+        return $this->isObjectType($node->class, $objectType);
+    }
+
+    /**
+     * @return array<int, Arg>
+     */
+    private function resolveNewArguments(
+        SwapMethodCallArguments $swapMethodCallArguments,
+        MethodCall|StaticCall $call
+    ): array {
+        $newArguments = [];
+        foreach ($swapMethodCallArguments->getOrder() as $oldPosition => $newPosition) {
+            if (! isset($call->getArgs()[$oldPosition])) {
```

Comment:
`call->getArgs()` in loop can be moved to before loop to a variable

Reply:
@samsonasik Could you handle the update and then merge it?

Reply:
Yes, updated https://github.com/rectorphp/rector-src/pull/3726/commits/01c5f058b6e9d2d646f93c4b19d6eebb453cf667